src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -518,6 +521,17 @@ class Manager : public DoutPrefixProvider {
           }
         }
       }
+
+      // updating perfcounters with topic stats
+      uint64_t entries_size;
+      uint32_t entries_number;
+      const auto ret = cls_2pc_queue_get_topic_stats(rados_ioctx, queue_name, entries_number, entries_size);
+      if (ret < 0) {
+        ldpp_dout(this, 1) << "ERROR: topic stats for topic: " << queue_name << ". error: " << ret << dendl;
+      }
+
+      queue_counters_container.set(l_rgw_persistent_topic_len, entries_number);
```

Comment:
if you fail to fetch the stats, we should probably not update the perf counter and keep the previous ones.